cabal-install/src/Distribution/Client/CmdStatus.hs

```diff
@@ -0,0 +1,367 @@
+        -- This error should never be shown
+        ("couldn't parse targets: " ++)
+        -- TODO: wrong parser, kills filepaths with spaces
+        (parsecCommaList parsecToken)
+
+-------------------------------------------------------------------------------
+-- Action
+-------------------------------------------------------------------------------
+
+-- | Entry point for the 'ide' command.
+statusAction :: NixStyleFlags StatusFlags -> [String] -> GlobalFlags -> IO ()
+statusAction flags@NixStyleFlags { extraFlags = statusFlags, ..} cliTargetStrings globalFlags = do
+  when (NoFlag == statusOutputFormat statusFlags) $ do
+    die' verbosity "The status command requires the flag '--output-format'."
+  when (not $ null cliTargetStrings) $
+    die' verbosity "The status command takes not target arguments directly. Use appropriate flags to pass in target information."
+
+  baseCtx <- establishProjectBaseContext verbosity cliConfig OtherCommand
+  (_, elaboratedPlan, elabSharedConfig, _, _) <-
+    rebuildInstallPlan verbosity
+                        (distDirLayout baseCtx)
+                        (cabalDirLayout baseCtx)
+                        (projectConfig baseCtx)
+                        (localPackages baseCtx)
+
+  let initialJson = Json.object
+        [ "cabal-version" .= jdisplay cabalInstallVersion
+        ]
+
+  compilerJson <- if not $ fromFlagOrDefault False (statusCompiler statusFlags)
+    then pure $ Json.object [] -- Neutral element
+    else do
+      let compiler = pkgConfigCompiler elabSharedConfig
+      compilerProg <- requireCompilerProg verbosity compiler
+      let progDb = pkgConfigCompilerProgs elabSharedConfig
+      (configuredCompilerProg, _) <- requireProgram verbosity compilerProg progDb
+      pure $ mkCompilerInfo configuredCompilerProg compiler
+
+  buildInfoJson <- if null (statusBuildInfo statusFlags)
+    then pure $ Json.object [] -- Neutral element
+    else do
+      let targetStrings = statusBuildInfo statusFlags
+      targetSelectors <- readTargetSelectors (localPackages baseCtx) Nothing targetStrings >>= \case
+        Left err   -> reportTargetSelectorProblems verbosity err
+        Right sels -> pure sels
+
+      -- Interpret the targets on the command line as build targets
+      -- (as opposed to say repl or haddock targets).
+      -- TODO: don't throw on targets that are invalid.
+      targets <- either (reportBuildTargetProblems verbosity) return
+                $ resolveTargets
+                    selectPackageTargets
+                    selectComponentTarget
+                    elaboratedPlan
+                    Nothing
+                    targetSelectors
+
+      pure $ mkBuildInfoJson (distDirLayout baseCtx) elabSharedConfig
+              elaboratedPlan targets targetSelectors targetStrings
+
+  let statusJson = mergeJsonObjects [initialJson, compilerJson, buildInfoJson]
+
+  -- Final output
+  putStrLn $ withOutputMarker verbosity $ Json.encodeToString statusJson
```

Comment:
Since  cabal is rather annoying wrt piping output to files (e.g. you have to be careful to suppress other stuff printing to stdout)... having a `--file` option or so may be worthwhile.

Reply:
That's basically what the marker is helpful for, as you can filter for the markers and get only the json output.

I am open to adding a file output, but are other parts of the API using file output options as well?